src/main/java/seedu/address/ui/CalendarView.java

```diff
@@ -0,0 +1,227 @@
+        dateLabel.setText(label);
+
+        // change date headers
+        dateHeaderMon.setText(calendarStartDate.format(dateFormatter));
+        dateHeaderTue.setText(calendarStartDate.plusDays(1).format(dateFormatter));
+        dateHeaderWed.setText(calendarStartDate.plusDays(2).format(dateFormatter));
+        dateHeaderThur.setText(calendarStartDate.plusDays(3).format(dateFormatter));
+        dateHeaderFri.setText(calendarStartDate.plusDays(4).format(dateFormatter));
+        dateHeaderSat.setText(calendarStartDate.plusDays(5).format(dateFormatter));
+        dateHeaderSun.setText(calendarStartDate.plusDays(6).format(dateFormatter));
+    }
+
+    /**
+     * Helper method to populate list of {@code Tuition} onto the Calendar's tuition list
+     */
+    private void populateTuitions() {
+        tuitionList.clear();
+        ObservableList<Student> studentObservableList = logic.getFilteredStudentList();
+        for (int i = 0; i < studentObservableList.size(); i++) {
+            for (int j = 0; j < studentObservableList.get(i).getListOfSessions().size(); j++) {
+                Student currStudent = studentObservableList.get(i);
+                Session currSession = currStudent.getListOfSessions().get(j);
+                tuitionList.add(new Tuition(currStudent, currSession, i, j));
+            }
+        }
+    }
+
+    /**
+     * Invokes listener method to update calendar's tuition list upon new updates
+     */
+    private void setStudentListListener() {
+        ObservableList<Student> studentObservableList = logic.getFilteredStudentList();
+        studentObservableList.addListener((ListChangeListener<Student>) change -> {
+            while (change.next()) {
+                populateTuitions();
+            }
+        });
+    }
+
+    /**
+     * Helper method for checking if session (individual and recurring) takes place on a particular date
+     */
+    private boolean hasSessionOnDate(LocalDateTime date, Tuition tuition) {
+        Session tuitionSession = tuition.getSession();
+        LocalDateTime tuitionDateTime = tuitionSession.getSessionDate().getDateTime();
+        boolean isSessionOnDate = tuitionDateTime.toLocalDate().isEqual(date.toLocalDate());
+        boolean isRecurringSessionOnDate = false;
+
+        if (tuitionSession instanceof RecurringSession) {
+            RecurringSession recurringTuitionSession = (RecurringSession) tuitionSession;
+            isRecurringSessionOnDate = recurringTuitionSession.hasSessionOnDate(new SessionDate(date));
+        }
+
+        return isSessionOnDate || isRecurringSessionOnDate;
+    }
+
+    /**
+     * Binds Monday - Sunday {@code ListView} to respective filtered and sorted tuition list
+     */
+    private void setListViews(ObservableList<Tuition> tuitionList) {
+
+        Comparator<Tuition> cmpSessionDate = Comparator.comparing(a -> a.getSession().getSessionDate().getTime());
```

Comment:
Hi, was just wondering is using sessionDateComparator instead of cmpSessionDate might be more meaningful. 

Replacing "a" for the "a -> a.." to something more meaningful will be helpful for the developers as well.

Reply:
actually used `cmpSessionDate` to shorten code 😆 
will make the change, for both `cmpSessionDate` and `a`!